test/e2e-tests.sh

```diff
@@ -57,7 +57,15 @@ initialize $@
 
 header "Setting up environment"
 
-install_knative_serving || fail_test "Knative Serving installation failed"
+if [[ "${RUN_GLOO_TESTS}" -eq "1" ]]; then
```

Comment:
1. Use a flag instead of an environment variable, it's better. Update `parse_flags()` in `e2e-common.sh`.
1. `install_knative_serving_gloo_version` looks like a copy of `install_knative_serving` with a few changes. It seems to me that you could call `install_knative_serving` with a few changes or parameters and make it install Gloo instead of Istio. Same comment about the uninstall function. They are already complex enough, so I want to avoid duplicating them as much as possible.